### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'e.QR_CODE')

### ISSUE THREAD

**Issue by samvandamme** [mebjas/html5-qrcode#584]
**Describe the bug**
After getting the camera's and when you run `html5QrCode.start`, I'm getting the error 
Safari: `TypeError: undefined is not an object (evaluating 'e.QR_CODE')`
Chrome: `TypeError: Cannot read properties of undefined (reading 'QR_CODE')`

**To Reproduce**
Using the "hard" version `import { Html5Qrcode } from "html5-qrcode"`

```javascript
startCertainCameraButtonClicked(event) {
    event.preventDefault()
    const cameraId = event.target.dataset["camera-id"]

    const html5QrCode = new Html5Qrcode(cameraDivId);
    html5QrCode.start(
      cameraId,
      {
        fps: 20, // Optional, frame per seconds for qr code scanning
        qrbox: { width: 250, height: 250 },  // Optional, if you want bounded box UI
        // formatsToSupport: [ Html5QrcodeSupportedFormats.QR_CODE ]
      },
      (decodedText, decodedResult) => {
        // do something when code is read
        console.log("decodedText:   " + decodedText);
        console.log("decodedResult: " + decodedResult);
      },
      (errorMessage) => {
        // parse error, ignore it.
        console.log("error message 1: " + errorMessage);
      })
    .catch((err) => {
      // Start failed, handle it.
      console.log("error message 2: " + err);
    });
  }
```

**Expected behavior**
Not getting the error, but filling the div with the camera stream.

**Desktop (please complete the following information):**
 - OS: Mac OSX 13.0
 - Browser Chrome & Safari
 - Version 2.2.8
 - Stimulus on Ruby on Rails

**Comment by gioggi** [comment]
Confirm

**Comment by tarmotalu** [comment]
Confirm

**Comment by mebjas** [comment]
Which version of the library/

**Comment by tarmotalu** [comment]
Hi! 

Version 2.3.7. I'm loading through importmap script type. Do I use it correctly or missing something? Loading html5-qrcode.min.js directly works as expected.

```
<script type="importmap">{
      "imports": {"html5-qrcode": "https://ga.jspm.io/npm:html5-qrcode@2.3.7/esm/index.js"}
}</script>
```

And use it as
```
import { Html5Qrcode } from "html5-qrcode"
...
let scanner = new Html5Qrcode("reader")

```

Which results
```
TypeError: undefined is not an object (evaluating 'e.QR_CODE')
ZXingHtml5QrcodeDecoder — zxing-html5-qrcode-decoder.js:6
Html5QrcodeShim — code-decoder.js:52
Html5Qrcode — html5-qrcode.js:104

```

**Comment by melvinTmk** [comment]
Hello, have you found any solutions?  @samvandamme

**Comment by samvandamme** [comment]
> Hello, have you found any solutions? @samvandamme

No, we started using another library: https://github.com/nimiq/qr-scanner

**Comment by sato11** [comment]
I think we need to accept that this library is not ready to be used through Import Maps. I did try to do my part tackling the problem to get it to work but the root is so complicated I had to give up. But let me share what I have investigated before two years will have passed soon since the first report.

The root cause of this issue is that `./third_party/zxing-js.umd.js` is not compliant with ES Module syntax which Import Maps rely on. [What it seem to export as `BarcodeFormat`](https://github.com/mebjas/html5-qrcode/blob/v2.3.8/third_party/zxing-js.umd.js#L23779) is actually [a `var` combined with some cryptic immediately invoked function pattern](https://github.com/mebjas/html5-qrcode/blob/v2.3.8/third_party/zxing-js.umd.js#L3848). I'm not a professional of ES Module system but I feel I'm ready to simply accept if someone tells me "this `var` cannot just be exported".

So `./third_party/zxing-js.umd.js` fails to export `BarcodeFormat`, which `./src/zxing-html5-qrcode-decoder.ts` wants to retrieve only to panic. In summary, this is the abbreviated version of what is going on around [here](https://github.com/mebjas/html5-qrcode/blob/v2.3.8/src/zxing-html5-qrcode-decoder.ts#L31C51-L31C78) that pops up the error in our browser with so much more esoteric language:

```javascript
>import * as ZXing from "../third_party/zxing-js.umd"
>ZXing.BarcodeFormat
undefined
>ZXing.BarcodeFormat.QR_CODE
Uncaught TypeError: ZXing.BarcodeFormat is undefined
```

That is to say we can fix the error if we can get `zxing-js.umd.js` which is fully compliant with ES Module. So we need to go to https://github.com/zxing-js/library and try building it. In fact the repo seems to contain the npm script named `build:esm` which should produce the artifact that just works, but it does not work like it's supposed to. Plus they say the project is in maintenance mode and things seem to be very inactive there, which shrinks a subtle hope even further and I had to give up. 

To look at the repository, a bunch of tsconfigs seem to be present, one of which being `tsconfig.lib-esm.json`, but [their module-resolution](https://github.com/zxing-js/library/blob/master/tsconfig.json#L6) targets `"node"`, which [apparently supports CommonJS alone](https://www.typescriptlang.org/tsconfig/#moduleResolution). The attempt to upgrade the Node.js was in vain. The attempt to upgrade TypeScript was in vain. Being nothing more than a part time JavaScripter, I admit that I have been defeated.

That's how I have come to conclude it is impossible at the moment to produce the `zxing-js.umd.js` which is compliant with ES Module. It means it is also impossible to execute `html5-qrcode` using Import Maps, in the browser, directly. Finding a way to build a valid `zxing-js.umd.js` out of zxing-js library should be a key, but zxing-js being quite legacy perhaps we won't be able to see the fix forever. Also, no simple workaround has been found.

Migrating to a different library that works in the browser is a great idea as long as it works. My team perhaps will reconsider the product requirements in order to give up the camera-in-browser approach, though.
